cedar-drt/fuzz/src/lib.rs

```diff
@@ -33,180 +31,158 @@ use cedar_policy_core::extensions::Extensions;
 pub use cedar_policy_validator::{ValidationErrorKind, ValidationMode, Validator, ValidatorSchema};
 use log::info;
 
-pub struct DifferentialTester<'e> {
-    /// Rust engine instance
-    authorizer: Authorizer,
-    /// Definitional engine instance
-    def_engine: DefinitionalEngine<'e>,
-    /// Definitional validator instance
-    def_validator: DefinitionalValidator<'e>,
+/// Compare the behavior of the evaluator in cedar-policy against a custom Cedar
+/// implementation. Panics if the two do not agree. `expr` is the expression to
+/// evaluate and `request` and `entities` are used to populate the evaluator.
+pub fn run_eval_test(
+    custom_impl: &dyn CedarTestImplementation,
```

Comment:
does this need to be `&dyn`?  here and in the similar methods.  would `&impl` just work as a drop-in replacement?

Reply:
`&impl` does indeed work as a drop-in replacement. Thanks for pointing that out.